internal/store/build_result.go

```diff
@@ -148,6 +148,14 @@ func (b BuildState) HasImage() bool {
 	return b.LastResult.HasImage()
 }
 
+// Whether the image represented by this state needs to be built.
+// If the image has already been built, and no files have been
+// changed since then, then we can re-use the previous result.
+func (b BuildState) NeedsImageBuild() bool {
+	alreadyBuilt := b.LastResult.HasImage()
+	return !alreadyBuilt || len(b.FilesChangedSet) > 0
```

Comment:
hmm will this respect config changes? I believe "this manifest is different now b/c of config file changes" is tracked somewhere besides `FilesChangedSet`.

Reply:
yes.

The BuildState needs to be the BuildState for this manifest. If the config files changed the manifest, then the BuildState is reset. see:

https://github.com/windmilleng/tilt/blob/master/internal/engine/upper.go#L480

(Note that other things use the FilesChangedSet, so if this was broken, this wouldn't be the first place it was breaking)